File: php/english-demo/util/xputil.php
<?php

    /**
     * hash sign
     * @param signSource - the string to sign
     * @return
     */
    function sign($signSource) {
        $signature = sha1($signSource);

        return $signature;
    }   

   	//get millisecond
	function getMillisecond() {
	    list($s1, $s2) = explode(' ', microtime());
	    return sprintf('%.0f', (floatval($s1) + floatval($s2)) * 1000);
	} 

?>

File: go/english/src/xpyun-opensdk/util/xputil.go
package util

import (
	"crypto/sha1"
	"fmt"
	"strings"
	"time"
)

/**
 * hash sign
 * @param signSource - the string to sign
 * @return
 */
func Sign(signSource string) string {
	h := sha1.New()
	h.Write([]byte(signSource))
	result := fmt.Sprintf("%x", h.Sum(nil))

	return result
}

//get millisecond
func GetMillisecond() int64 {
	return time.Now().UnixNano() / 1e6
}

/**
 * repeat the string to specific times
 * @param str - the string to repeat
 * @param repeatTimes - repeat times
 */
func StrRepeat(str string, repeatTimes int) string {
	return strings.Repeat(str, repeatTimes)
}
